dateutil/tz/tz.py

```diff
@@ -191,7 +205,7 @@ def dst(self, dt):
 
     @tzname_in_python2
     def tzname(self, dt):
-        return time.tzname[self._isdst(dt)]
+        return _gettznames()[self._isdst(dt)]
```

Comment:
**Note to reviewers**: there is a performance overhead to decoding the timezone names on the fly.  My initial attempt cached the decoded values to perform it only once, but this broke some tests which mock this.  I preferred leaving the test suite unchanged, but if you're worried about performance, suggestions for improvements are welcome :-)